CRM/Utils/System/Joomla.php

```diff
@@ -578,10 +586,17 @@ public function loadBootStrap($params = [], $loadUser = TRUE, $throwError = TRUE
     }
 
     // Get the framework.
-    if (file_exists($joomlaBase . '/libraries/import.legacy.php')) {
+    if (file_exists($joomlaBase . '/libraries/import.legacy.php') && version_compare(JVERSION, '4.0', '<')) {
       require $joomlaBase . '/libraries/import.legacy.php';
     }
-    require $joomlaBase . '/libraries/cms.php';
+
+    if (version_compare(JVERSION, '4.0', '<')) {
+      require $joomlaBase . '/libraries/cms.php';
+    }
+    else {
+      require $joomlaBase . '/libraries/bootstrap.php';
+      require_once $joomlaBase . '/includes/framework.php';
+    }
     self::getJVersion($joomlaBase);
```

Comment:
@seamuslee001 can you move this function call to L577 above? That will prevent the internal notices : 
```sh

Warning: Use of undefined constant JVERSION - assumed 'JVERSION' (this will throw an Error in a future version of PHP) in /Applications/MAMP/htdocs/joomla/administrator/components/com_civicrm/civicrm/CRM/Utils/System/Joomla.php on line 589

Deprecated: Bootstrapping Joomla using the /Applications/MAMP/htdocs/joomla4/libraries/import.legacy.php file is deprecated.  Use /Applications/MAMP/htdocs/joomla4/libraries/bootstrap.php instead. in /Applications/MAMP/htdocs/joomla/libraries/import.legacy.php on line 16

Warning: Use of undefined constant JVERSION - assumed 'JVERSION' (this will throw an Error in a future version of PHP) in /Applications/MAMP/htdocs/joomla4/administrator/components/com_civicrm/civicrm/CRM/Utils/System/Joomla.php on line 593
```

Reply:
@monishdeb good point updated that now